### PR TITLE
Add `as tuple`

### LINKED ISSUE #1037: Add `as tuple`
`as const` makes tuple types `readonly`:

```typescript
const arr = [1, 2, 3] as const;
//    ^? const arr: readonly [1, 2, 3]
```

There’s a hack to solve this using `satisfies`:

```typescript
const arr = [1, 2, 3] satisfies [unknown, ...unknown[]];
//    ^? const arr: [1, 2, 3]
```

Maybe Civet could have `as tuple` as shorthand for `satisfies [unknown, ...unknown[]]`?

This solves issue #1004

### PR DESCRIPTION
**Pull request by bbrk24**
Closes #1037

### CODE REVIEW

**Comment by somebody1234** [comment]
- probably doesn't work for length-0 tuples
- probably doesn't work for readonly arrays (`as const`).

```ts
_ satisfies readonly unknown[] | []
```

might be better.

[See in TS Playground](https://www.typescriptlang.org/play?#code/NoXQBAzghgLglhAZnAphMwCuA7A1tgewHdsAaMAOip32O1BAChRJYFk0wbCSWAfDExZR0AYwLYIMVvCSp03Ov0GNxk6UQAWsMAF4MARnAiwaqYy07osjgrw964AaAvbp19vK72lTsACcUKAATCQAbAE8VS3c2OU5AkPCoxV4-UCA)

**Comment by bbrk24** [comment]
For some reason I thought that empty array automatically was inferred as `[]`. But you have a point, I'll update the `Tuple` token to reflect this.

File: source/parser.hera [hunk 1]
```diff
@@ -444,7 +444,16 @@ TypePostfix
   _:ws NWTypePostfix:postfix ->
     return [ws, ...postfix]
 
+Tuple
+  "tuple" ->
```
Comment (edemaine) [inline source/parser.hera:448]:
```suggestion
  "tuple" NonIdContinue ->
```

This is necessary to e.g. prevent matching `as tuples`

**Review by edemaine — APPROVED** [review]
Looks good!  Tested [this playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAHjAvDA2gRgDQwEwF0YQCGUAlhAGYkCmEMATlYQCbgA2AnjAK5gDWYIAO5gU+AD6pcAKBiyYAenkA9APxTQkWJ2ToseGIVoboBYmUo16jFmA7c+A4aJgTRMuYtVA).

**Pull request merged**